### DISCUSSION TITLE
support macos options(aka alt) key to work well with zellij

### DISCUSSION THREAD

**Discussion by qballer** [ghostty-org/ghostty#3483] [Q&A] [answered]
OS: MacOS 13.6.6 
multiplexer: [zellij](https://github.com/zellij-org/zellij/)

When using Zellij with Ghostty, the alt key is mapped to the wrong action. I'm unsure which mapping is being used, but this appears to be a bug in Ghostty. I'm going to review the docs to see how I can fix this, but if the maintainers can help, I would really appreciate it. 

Recreate:
=============
cargo install zellij
zellij 
ctrl+p  -> n
options+arrow keys open a floating window instead of moving between panes. 
related to: https://github.com/zellij-org/zellij/issues/265

**Comment by sadotsoy** [comment]
did you find a possible fix? @qballer

**Reply by sadotsoy** [reply]
@qballer https://github.com/ghostty-org/ghostty/discussions/3207

**Comment by qballer** [comment]
These are my current configuration 
```
keybind = alt+left=unbind
keybind = alt+right=unbind
macos-option-as-alt = true
```
And it doesn't seem to solve the issue, any ideas on whats the issue here?

**Reply by qballer** [reply]
issue seems to work

**Reply by guilhermeramosht** [reply]
Worked for me 🚀

**Comment by trskare** [comment]
I use ALT (super in ghostty) as a leader key for my zellij keybinds. This seems to crash with a lot ghostty´s default keybindings (you can see them by running `ghostty +list-keybinds`. I solved this by setting `keybind=clear` in the ghostty config file. I have to find other keybindings for the necessary  ghostty actions. I will not use a lot of them anyway because i use zellij.

**Reply by Gelio** [reply]
Thanks! That worked out for me. I used this solution to clear the Alt-Right and Alt-Left keybinds so they can be used in Neovim by https://github.com/github/copilot.vim

**Comment by peteringram0** [comment]
Also seeing this issue. For me i would love to get `ALT+K` working as thats second nature for my workflow.

**Comment by aldanor** [comment]
Same here. In zellij, option + left/right arrow within zellij doesn't work and triggers other things.

**Comment by sadotsoy** [comment]
This worked for me:

```
macos-option-as-alt = true

keybind = alt+left=unbind
keybind = alt+right=unbind
```

**Reply by aldanor** [reply]
> This worked for me:
> 
> ```
> macos-option-as-alt = true
> 
> keybind = alt+left=unbind
> keybind = alt+right=unbind
> ```

Can confirm, this works (perhaps it should be documented better somewhere?)

**Reply by citypaul** [reply]
This isn't working for me for some reason :(

**Comment by dunxiii** [comment, marked as answer]
On nordic keyboards we often use option for pipe and at characters, so I did only a set left option to act as alt. Works great.
```
macos-option-as-alt = left

keybind = alt+left=unbind
keybind = alt+right=unbind
```

**Reply by RashidRysaev** [reply]
Perfect, thanks for this one

**Reply by boern** [reply]
Thx,that work well,great!

**Reply by gnzsnz** [reply]
> macos-option-as-alt = left
> 
> keybind = alt+left=unbind
> keybind = alt+right=unbind

This solves byobu key bindings too 🚀

**Reply by petkostas** [reply]
Thanks! you reduced my cognitive load today! :)

**Reply by andirkh** [reply]
thanks !

**Reply by WHYBBE** [reply]
Thanks! It works.

**Reply by kirhgoff** [reply]
Bless you, thank you so much

**Reply by expert** [reply]
Thanks!

**Reply by PapaniaP** [reply]
Over a year later and this saved me from going into a mental asylum <3

**Reply by madsboddum** [reply]
Thanks for posting this.

**Comment by giacomoran** [comment]
I rely quite a bit on `Option+Left` and `Option+Right` to move the cursor one word at a time in the terminal, which stops working when setting
```
keybind = alt+left=unbind
keybind = alt+right=unbind
```

I solved this by:
- Setting only `macos-option-as-alt = true` in Ghostty
- Replacing the `Alt+ArrowKey` bindings in Zellij with `Alt+Shift+ArrowKey` to move between panes and tabs
- Removing the `Alt+F` shortcut in Zellij that opens the floating pane
